runtime/src/iree/builtins/ukernel/arch/arm_64/CMakeLists.txt

```diff
@@ -4,8 +4,19 @@
 # See https://llvm.org/LICENSE.txt for license information.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-check_cxx_compiler_flag("-march=armv8.2-a+dotprod" IREE_UK_BUILD_ARM_64_DOTPROD)
-check_cxx_compiler_flag("-march=armv8.2-a+i8mm" IREE_UK_BUILD_ARM_64_I8MM)
+iree_select_compiler_opts(IREE_UK_COPTS_ARM_64_DOTPROD
+  CLANG_OR_GCC
+    "-march=armv8.2-a+dotprod"
+)
+
+iree_select_compiler_opts(IREE_UK_COPTS_ARM_64_I8MM
+  CLANG_OR_GCC
+    "-march=armv8.2-a+i8mm"
+)
+
+check_cxx_compiler_flag("${IREE_UK_COPTS_ARM_64_DOTPROD}" IREE_UK_BUILD_ARM_64_DOTPROD)
```

Comment:
we need to get rid of these checks and move them into the code
when we are cross-compiling for embedding the library in the compiler we won't be using the cmake default compiler

every line of cmake around this is a big cost we'll need to duplicate both here, in bazel, and in the configuration for embedded linking - an ifdef in code is worth 1000x the weight of a cmake check in gold :)

can defer it to a follow-up, but it's a blocker to getting the ukernels into the compiler

Reply:
Oh although for this purpose I suppose that we can only use a host compiler if it's a Clang, so we would only need to implement version checks for Clang specifically ?